jdaviz/configs/imviz/plugins/rotate_image/rotate_image.py

```diff
@@ -0,0 +1,35 @@
+import math
+
+from traitlets import Any
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin, ViewerSelectMixin
+
+
+@tray_registry('imviz-rotate-image', label="Simple Image Rotation")
+class RotateImageSimple(TemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "rotate_image.vue"
+
+    angle = Any(0).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._theta = 0  # degrees, clockwise
+
+    def vue_rotate_image(self, *args, **kwargs):
+        # We only grab the value here to avoid constantly updating as
+        # user is still entering or updating the value.
+        try:
+            self._theta = float(self.angle)
+        except Exception:
+            return
+
+        viewer = self.app._viewer_by_id(self.viewer_selected)
+
+        # Rotate selected viewer canvas.
+        # TODO: Translation still a bit broken if zoomed in.
+        viewer.state.rotation = math.radians(self._theta)

```

Comment:
Works but pretty sure it is clockwise again... Is this expected?

Also, the center translation is still a bit off when I zoom in first and then rotate on a rectangular image. If I zoom in on a star, I would expect rotation around the star. But the star moves quite dramatically when I rotate.

Reply:
Apparently applying the transform directly on `x`, `y` in the `pretransform.__call__()` seems to have it operate the other way around again, but that is easily fixed, even if I cannot follow the full chain of events.

I think for finding the right centre of rotation setting `xy` from `(x_min + x_max) / 2` rather than `shape[0] / 2` etc. is still the better option (`x_min`, `y_min` may be nonzero even if not zoomed in); at least that that keeps the viewport centred and rotated around the original zoom region _when zooming in first and then rotating_. Unfortunately with the same modification, zooming in _after rotation_ is completely off – apparently this is setting the limits from the unrotated frame, I recon this is failing due to the data values not being returned from the transformed pixel coordinates just as with the mouseover.
I would expect that once the latter problem is solved, the zoom will also operate on the correct subset, so the alternative above still looks a step forward.

Reply:
Derek, all this will be fixed upstream, yes?

Reply:
Not sure which of the "all" is fixable upstream. I have a patch for the sense of direction, and keeping the centre of rotation in the current viewport centre, but the mouseover stats, active selection area and zoom-after-rotation remain of.

> * We should probably provide a method in glue-core on the state class which given cursor coordinates can provide the actual array index into the data since these are now no longer the same - we can then use that to fix the mouse over coordinates here

I think I'll work on that next, but it sounds like this will still require some adaption from Imviz to use those `viewer.state` methods.

Reply:
If it helps, the Imviz mouseover logic is here:

https://github.com/spacetelescope/jdaviz/blob/c2a9b82d52a470c22f79a89babe59268219e617c/jdaviz/configs/imviz/plugins/viewers.py#L97

Out of scope now but if you are curious, Cubeviz version is here:

https://github.com/spacetelescope/jdaviz/blob/c2a9b82d52a470c22f79a89babe59268219e617c/jdaviz/configs/cubeviz/plugins/viewers.py#L65

They both send info to the same plugin here:

https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/imviz/plugins/coords_info/coords_info.py

Reply:
Yes; looked at the above, but can I find any more info on what type of object the `data` is that is passed to `on_mouse_or_key_event`? I've been struggling to find out what its `'domain'` item is or where it is set.

Reply:
That I am not sure... Could be glue-jupyter , bqplot-image-gl , bqplot ... 🤷 

Hopefully @maartenbreddels or @astrofrog can tell us.